gateway/proxy_muxer.go

```diff
@@ -439,43 +441,21 @@ func (m *proxyMux) serve(gw *Gateway) {
 		case "tcp", "tls":
 			mainLog.Warning("Starting TCP server on:", p.listener.Addr().String())
 			go p.tcpProxy.Serve(p.getListener())
-		case "http", "https", "h2c":
-			mainLog.Warning("Starting HTTP server on:", p.listener.Addr().String())
-			readTimeout := 120 * time.Second
-			writeTimeout := 120 * time.Second
-
-			if conf.HttpServerOptions.ReadTimeout > 0 {
-				readTimeout = time.Duration(conf.HttpServerOptions.ReadTimeout) * time.Second
-			}
-
-			if conf.HttpServerOptions.WriteTimeout > 0 {
-				writeTimeout = time.Duration(conf.HttpServerOptions.WriteTimeout) * time.Second
-			}
-
-			h := &handleWrapper{
-				router:             p.router,
-				maxRequestBodySize: conf.HttpServerOptions.MaxRequestBodySize,
-			}
-
-			// by default enabling h2c by wrapping handler in h2c. This ensures all features including tracing work
-			// in h2c services.
-			h2s := &http2.Server{}
-			handler := &h2cWrapper{
-				w: h,
-				h: h2c.NewHandler(h, h2s),
-			}
-
-			addr := conf.ListenAddress + ":" + strconv.Itoa(p.port)
-			p.httpServer = &http.Server{
-				Addr:         addr,
-				ReadTimeout:  readTimeout,
-				WriteTimeout: writeTimeout,
-				Handler:      handler,
+		case "https", "h2c":
+			mainLog.Warning("Starting HTTPS server on:", p.listener.Addr().String())
+			p.SetupServer(gw)
+			listener, err := UpgradeHTTPListenerToHTTPS(p.port, p.getListener(), gw)
+			if err != nil {
+				mainLog.WithError(err).Error("Can't start listener")
+				continue
 			}
-			if gw.ConnectionWatcher != nil {
```

Comment:
This likely shouldn't have been removed, it's use case is to track the connection state

Reply:
check returned